### PR TITLE
Allow fetching public data sources and data source views.

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

With the introduction of the public vault concept, we aim to use public data sources. However, in practice, this isn't currently possible because all SQL queries are restricted to the current workspace, preventing access to public data sources from other workspaces.

This PR removes the SQL clause that limits queries by workspace ID in all cases of fetching by ID. Instead, it introduces a `canFetch` method to ensure that either the vault is public or the user is a member of the workspace.

I took the opportunity to refactor the `canList` method to make it more robust.

## Risk

Worst case it breaks the listing of data sources, data source views and apps. Safe to rollback.

### CODE REVIEW

File: front/lib/resources/data_source_view_resource.ts [hunk 1]
```diff
@@ -204,12 +204,15 @@ export class DataSourceViewResource extends ResourceWithVault<DataSourceViewMode
...
+        where: {
+          workspaceId: auth.getNonNullableWorkspace().id,
+        },
+      }
     );
 
-    return dataSourceViews.filter(
-      (dsv) => auth.isAdmin() || auth.hasPermission([dsv.vault.acl()], "read")
-    );
+    return dataSourceViews.filter((dsv) => dsv.canList(auth));
```
Comment (Fraggle) [inline front/lib/resources/data_source_view_resource.ts:215]: I understand your intent but the change is not a 1-1 change i believe. You may break something.

**Review by Fraggle — APPROVED** [review]
LGTM

File: front/lib/resources/data_source_resource.ts [hunk 2]
```diff
@@ -286,18 +297,9 @@ export class DataSourceResource extends ResourceWithVault<DataSourceModel> {
     auth: Authenticator,
     options?: FetchDataSourceOptions
   ): Promise<DataSourceResource[]> {
-    return this.baseFetchWithAuthorization(auth, this.getOptions(options));
-  }
-
-  static async listByWorkspaceIdAndNames(
```
Comment (flvndvd) [inline front/lib/resources/data_source_resource.ts:292]: This method was not used anymore.

**Review by tdraier — APPROVED** [review]

File: front/lib/resources/data_source_resource.ts [hunk 3]
```diff
@@ -257,8 +268,7 @@ export class DataSourceResource extends ResourceWithVault<DataSourceModel> {
     names: string[],
     options?: Omit<FetchDataSourceOptions, "limit" | "order">
   ): Promise<DataSourceResource[]> {
-    const dataSources = await this.baseFetchWithAuthorization(auth, {
-      ...this.getOptions(options),
+    const dataSources = await this.baseFetch(auth, options, {
       where: {
         name: {
           [Op.in]: names,
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/lib/resources/data_source_resource.ts:274]: Good catch, added.

**Pull request merged**